flake.nix

```diff
@@ -15,11 +15,39 @@
       let
         overlays = [ (import rust-overlay) ];
         pkgs = import nixpkgs { inherit system overlays; };
+        lib = pkgs.lib;
+
         naersk' = pkgs.callPackage naersk {
           cargo = pkgs.rust-bin.stable.latest.default;
           rustc = pkgs.rust-bin.stable.latest.default;
         };
-      in {
+
+        buildXwaylandSatellite = { withSystemd ? false }: naersk'.buildPackage {
+          src = ./.;
+
+          nativeBuildInputs = with pkgs; [
+            rustPlatform.bindgenHook
+            rust-bin.stable.latest.default
+            pkg-config
+
+            xcb-util-cursor
+            xorg.libxcb
+
+            makeWrapper
+          ] ++ lib.optional withSystemd pkgs.systemd;
```

Comment:
Why is `systemd` added as a build input? `xwayland-satellite` only uses the `sd_notify` crate, which doesn't have any native dependency. [It writes to a text-based socket that systemd provides](https://github.com/lnicola/sd-notify/blob/6990e3733f69e5b23dad8985d66213dfb31675b6/src/lib.rs#L148-L151) and does little else; this dependency is pointless (and for instance, nixpkgs does not have it)